### PR TITLE
Setup associations

### PR DESCRIPTION
**Pull request by AyoMoses1**
I have been able to 
- Create model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd_v1_1.png).
- Set up associations between models.
Please feel free to review and drop your feedbacks where necessary

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :likes
+  has_many :comments
+
+  def update_user_post_counter
+    author.increment(:postscounter)
+  end
```
Comment (codepantha) [inline app/models/post.rb]: - [ ] [OPTIONAL] Great job implementing the `update_user_post_counter` method. A way to make it even cooler would be to use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the #update_user_post_counter` method every time a new post is created and saved to the database, then make the update_user_post_counter method private
### HINT 👇 
```ruby
after_save :update_post_counter

  private

  def update_user_post_counter
    author.increment!(:postscounter)
  end
```

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :post
+  belongs_to :post, class_name: 'User'
+
```
Comment (codepantha) [inline app/models/like.rb]: - [ ] I believe here you actually wanted to say `belongs_to :author, class_name: 'User`. Kindly fix the error 🙏

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def update_post_comment_counter
+    post.increment(:commentscounter)
+  end
```
Comment (codepantha) [inline app/models/comment.rb:11]: - [ ] [OPTIONAL] Please consider using one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_post_comment_counter` method every time a new comment is created and saved to the database, then make the update_post_comment_countermethod private

File: app/models/like.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+  def update_post_like_counter
+    post.increment(:likescounter)
+  end
```
Comment (codepantha) [inline app/models/like.rb:11]: - [ ] [OPTIONAL] Please consider using one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_post_like_counter` method every time a new post is created and saved to the database, then make the update_post_like_counter method private

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @AyoMoses1 👋 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 🎯 
✔️ Descriptive PR
✔️ Create and set up associations between models 📌 
✔️ Good commit messages
✔️ No linter errors ✅ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by MeqdamAlqudah — APPROVED** [review]
Hi @AyoMoses1 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
image

## Highlights 
:+1: models relations 
:+1: files structure

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**